### ISSUE TITLE
compile failed

### ISSUE THREAD

**Issue by danhantao** [haxpor/Potatso#2]
Fetching podspec for `Aspects` from `./Library/Aspects/`
Fetching podspec for `ICSMainFramework` from `./Library/ICSMainFramework/`
[!] Unable to integrate the following embedded targets with their respective host targets (a host target is a "parent" target which embeds a "child" target like a framework or extension):

- Potatso (3.0) and PacketProcessor () do not use the same Swift version.
- Potatso (3.0) and PacketTunnel () do not use the same Swift version.
- PotatsoLibraryTests () and PotatsoLibrary (3.0) do not use the same Swift version.

[!] There are duplicate dependencies on `SwiftColor` in `Podfile`:

- SwiftColor (~> 0.3.7)
- SwiftColor

**Comment by haxpor** [comment]
Hi @danhantao thanks for reporting.
Could you give me your cocoapod version, and steps you took to have above result please?

**Comment by danhantao** [comment]
Thanks for your reply!

cocoapod version:1.1.1
excute:pod install
result:
Fetching podspec for `Aspects` from `./Library/Aspects/`
Fetching podspec for `ICSMainFramework` from `./Library/ICSMainFramework/`
[!] Unable to integrate the following embedded targets with their respective host targets (a host target is a "parent" target which embeds a "child" target like a framework or extension):

- Potatso (3.0) and PacketProcessor () do not use the same Swift version.
- Potatso (3.0) and PacketTunnel () do not use the same Swift version.
- PotatsoLibraryTests () and PotatsoLibrary (3.0) do not use the same Swift version.

[!] There are duplicate dependencies on `SwiftColor` in `Podfile`:

- SwiftColor (~> 0.3.7)
- SwiftColor

**Comment by haxpor** [comment]
Thanks! Personally I use `1.0.1`. I will test it against newer version and report back.

**Comment by haxpor** [comment]
Update: Found the problem. It happens when using cocoapod version `1.1.1`. I'll roll out the fix soon.

**Comment by danhantao** [comment]
<img width="1133" alt="screen shot 2017-01-12 at 11 23 20 am" src="https://cloud.githubusercontent.com/assets/5282985/21875810/a61fa362-d8b9-11e6-825b-7d36ca878445.png">
Thanks for your help!
when I compile it by your steps,compile failed.

Xcode Version:Version 8.2.1 (8C1002)
Mac:10.12.2 (16C67)
Cocoapod:1.0.1

**Comment by haxpor** [comment]
@danhantao `ShadowPath.h` cannot be found, for this problem, please check that you execute `git submodule update --init`. `ShadowPath` is as part of git submodule, we need to fetch it first.

**Comment by danhantao** [comment]
@haxpor how to set SWIFT_VERSION for ICSMainFrameword and KissXML

**Comment by haxpor** [comment]
@danhantao as I tested against cocoapods `1.1.1` I found that some of pods are still based on swift 2. I will fix this too in next update. Anyway, you have 2 options.

1. Remove specific version in `Podfile`.
   From `pod 'ICDMaterialActivityIndicatorView', '~> 0.1.0'` to `pod 'ICDMaterialActivityIndicatorView'', then execute `pod update` again.

2. In Xcode, click on Edit-> Convert -> To Current Swift Syntax, then choose those framework.

Both ways in the background will set `SWIFT_VERSION` to `3.0`.
